### PR TITLE
Windows install fixes

### PR DESCRIPTION
**Pull request by albertvaka**
- Fix source for latest. It was installing agent 5.
- Allow upgrades to happen by passing the version. Otherwise if any version is installed it noops.
- Fix blacklist check.

### CODE REVIEW

File: manifests/windows/agent6.pp [hunk 1]
```diff
@@ -16,8 +16,13 @@
   Enum['present', 'absent'] $ensure = 'present',
 ) inherits datadog_agent::params {
 
-  $msi_full_path = "${msi_location}/datadog-agent-6-${agent_version}.amd64.msi"
-  $msi_source = "${baseurl}ddagent-cli-${agent_version}.msi"
+  if $agent_version == "latest" {
+    $msi_source = "${baseurl}datadog-agent-6-latest.amd64.msi"
```
Comment (truthbk) [inline manifests/windows/agent6.pp]: wondering if we should maybe use `join` to build the URIs?

Comment (albertvaka) [reply manifests/windows/agent6.pp]: I haven't found a version of `join` specific for URLs, so I don't think it's going to be better than this actually.

File: manifests/windows/agent6.pp [hunk 2]
```diff
@@ -31,15 +36,21 @@
...
 
+    if $agent_version == 'latest' {
+      $ensure_version = 'installed'
+    } else {
+      $ensure_version = $agent_version
+    }
+
     package { $datadog_agent::params::package_name:
-      ensure          => installed,
+      ensure          => $ensure_version,
```
Comment (truthbk) [inline manifests/windows/agent6.pp]: Does this actually do something on windows?

File: manifests/windows/agent6.pp [hunk 3]
```diff
@@ -31,15 +36,21 @@
     }
 
     exec { 'validate':
-      command   => "\$blacklist = '928b00d2f952219732cda9ae0515351b15f9b9c1ea1d546738f9dc0fda70c336','78b2bb2b231bcc185eb73dd367bfb6cb8a5d45ba93a46a7890fd607dc9188194';\$fileStream = [system.io.file]::openread('${msi_full_path}'); \$hasher = [System.Security.Cryptography.HashAlgorithm]::create('sha256'); \$hash = \$hasher.ComputeHash(\$fileStream); \$fileStream.close(); \$fileStream.dispose();\$hexhash = [system.bitconverter]::tostring(\$hash).ToLower().replace('-','');if (\$hexhash -match \$blacklist) { Exit 1 }",
+      command   => "\$blacklist = '928b00d2f952219732cda9ae0515351b15f9b9c1ea1d546738f9dc0fda70c336','78b2bb2b231bcc185eb73dd367bfb6cb8a5d45ba93a46a7890fd607dc9188194';\$fileStream = [system.io.file]::openread('${msi_full_path}'); \$hasher = [System.Security.Cryptography.HashAlgorithm]::create('sha256'); \$hash = \$hasher.ComputeHash(\$fileStream); \$fileStream.close(); \$fileStream.dispose();\$hexhash = [system.bitconverter]::tostring(\$hash).ToLower().replace('-','');if (\$blacklist.Contains(\$hexhash)) { Exit 1 }",
```
Comment (truthbk) [inline manifests/windows/agent6.pp]: 👍

File: manifests/windows/agent6.pp (hunk 2 above)
Comment (albertvaka) [reply manifests/windows/agent6.pp]: If you set it to "installed", it won't upgrade it as long as it's installed. When you pass a version, it checks the installed version vs version the MSI and installs if they are different.

**Pull request merged**